File: src/automatica.core/Automatica.WebNew/src/app/base/model/node-data-type.ts
import { BaseModel, JsonFieldInfo, JsonProperty, Model } from "./base-model"

export enum NodeDataTypeEnum {
    NoAttribute = 0,
    Integer = 1,
    Double = 2,
    String = 3,
    Boolean = 4,
    DateTime = 5,
    Time = 6,
    Date = 7,
    WindowState = 8
}

@Model()
export class NodeDataType extends BaseModel {

    @JsonProperty()
    Type: number;

    @JsonProperty()
    Name: string;

    @JsonProperty()
    Description: string;

    NodeDataTypeEnum: NodeDataTypeEnum;

    protected afterFromJson(): void {
        this.NodeDataTypeEnum = <NodeDataTypeEnum>this.Type;
    }


    protected getJsonProperty(): Map<string, JsonFieldInfo> {
        return void 0;
    }

    public typeInfo(): string {
        return "NodeDataType";
    }
}

File: src/automatica.core/Automatica.Core.Base/Templates/NodeDataType.cs
using Automatica.Core.EF.Models;

namespace Automatica.Core.Base.Templates
{
    /// <summary>
    /// Available NodeDataTypes
    /// </summary>
    public enum NodeDataType
    {
        [NodeDataTypeEnum("NoAttribute")]
        NoAttribute = 0,
        [NodeDataTypeEnum("Integer")]
        Integer = 1,
        [NodeDataTypeEnum("Double")]
        Double = 2,
        [NodeDataTypeEnum("String")]
        String = 3,
        [NodeDataTypeEnum("Boolean")]
        Boolean = 4,
        [NodeDataTypeEnum("DateTime")]
        DateTime = 5,
        [NodeDataTypeEnum("Time")]
        Time = 6,
        [NodeDataTypeEnum("Date")]
        Date = 7,
        [NodeDataTypeEnum("WindowState")]
        WindowState = 8
    }
}
